### DISCUSSION TITLE
Writing CLI tools for Vendure

### DISCUSSION THREAD

**Discussion by skid** [vendurehq/vendure#1050] [Q&A] [answered]
How do I go about writing CLI tools for Vendure. The bootstrap method starts a server, but I need some container that will load the Vendure config and allow me to import and run custom code (e.g. run a query against the DB from the command line) ? Is there docs on this topic?

**Comment by michaelbromley** [comment, marked as answer]
(copying reply from Slack)

When you call `bootstrap(config)` it returns an instance of the Nest application, which you can use to get any provider from the app and use directly in your CLI program, e.g.:

```ts
const app = await bootstrap(config);
const connection = app.get(TransactionalConnection);
```

More can be found in the Nest docs: https://docs.nestjs.com/standalone-applications#standalone-applications

**Reply by skid** [reply]
This still starts a Vendure server. Is there a way to do it without running a server (mostly to rediuce wait times)

**Reply by michaelbromley** [reply]
You can use `bootstrapWorker()` in that case, which does not include the ApiModule and does not start a server.
